### PR TITLE
Search Teeing server side [WIP]

### PR DESCRIPTION
**Pull request by stkem**
An endpoint on shoebox to search which forwards the request to two search machines and returns the fastest.
Admin overview will show performance comparison of this with regular direct search.
RFC @raymondng 
cc @eishay @ymatsuda @2is10

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/common/service/ServiceClient.scala [hunk 1]
```diff
@@ -92,4 +92,9 @@ trait ServiceClient extends Logging {
       log.info(s"[broadcast] Sending to $url: ${body.toString.take(120)}")
       callUrl(call, url, body)
     }
+
+  protected def tee(call: ServiceRoute, body: JsValue = JsNull, teegree: Int = 2): Future[ClientResponse] = {
+    val futures = urls(call.url).take(teegree).map(callUrl(call, _, body))
```
Comment (eishay) [inline kifi-backend/modules/common/app/com/keepit/common/service/ServiceClient.scala]: nice :-)

File: kifi-backend/modules/search/app/com/keepit/controllers/ext/ExtSearchController.scala [hunk 2]
```diff
@@ -147,7 +147,52 @@ class ExtSearchController @Inject() (
...
+    tz: Option[String] = None,
+    coll: Option[String] = None) = Action { request =>
+
+    val res = doSearch(request, Id[User](userId), acceptLangs.split(","), noSearchExperiments, query, filter, maxHits, lastUUIDStr, context, kifiVersion, start, end, tz, coll)
+
+    Ok(res).withHeaders("Cache-Control" -> "private, max-age=10")
+  }
+
+
+  @Trace
```
Comment (eishay) [inline kifi-backend/modules/search/app/com/keepit/controllers/ext/ExtSearchController.scala]: does @Trace work?

**Comment by eishay** [comment]
you're fast :-)

File: kifi-backend/modules/common/app/com/keepit/common/service/ServiceClient.scala [hunk 3]
```diff
@@ -92,4 +92,9 @@ trait ServiceClient extends Logging {
...
+    Future.firstCompletedOf(futures)
```
Comment (eishay) [inline kifi-backend/modules/common/app/com/keepit/common/service/ServiceClient.scala:103]: nice, didn't know about that one

File: kifi-backend/modules/search/app/com/keepit/controllers/ext/ExtSearchController.scala (hunk 2 above)
Comment (stkem) [inline kifi-backend/modules/search/app/com/keepit/controllers/ext/ExtSearchController.scala]: No idea. It was actually there before and I thought it best not to remove without being sure what it does.

**Comment by raymondng** [comment]
+1 :) At a wedding will need to review later. Good work!

File: kifi-backend/modules/search/app/com/keepit/controllers/ext/ExtSearchController.scala (hunk 2 above)
Comment (ymatsuda) [inline kifi-backend/modules/search/app/com/keepit/controllers/ext/ExtSearchController.scala]: I don't know who added it to ExtSearchController.

Comment (yingjieMiao) [inline kifi-backend/modules/search/app/com/keepit/controllers/ext/ExtSearchController.scala]: The Trace thing was from NewRelic, I believe. 

We can remove it.

**Pull request merged**